gulpfile.js

```diff
@@ -1,12 +1,37 @@
 const gulp = require("gulp");
 const sass = require("gulp-sass")(require("sass"));
 const sourcemaps = require("gulp-sourcemaps"); // Add gulp-sourcemaps
+const ts = require("gulp-typescript");
+const uglify = require("gulp-uglify");
 
+// Create a TypeScript project
+const tsProject = ts.createProject("tsconfig.json");
+
+// Gulp task to compile TypeScript
+gulp.task("scripts", () => {
+  // Compile TypeScript files from src to dist
+  return tsProject
+    .src()
+    .pipe(tsProject()) // Compile TypeScript files
+    .pipe(uglify()) // Minify the output JavaScript
+    .pipe(gulp.dest("dist/js")); // Output to the 'dist' folder
```

Comment:
exports is not defined 
![image](https://github.com/user-attachments/assets/062b17aa-95a5-4fab-8b7a-1a1666406873)